### PR TITLE
Add vSphere paravirtual cloud provider template manifest

### PR DESCRIPTION
**Pull request by nicolehanjing**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add vSphere paravirtual cloud provider template manifest
```

### CODE REVIEW

**Comment by nicolehanjing** [comment]
cc @lubronzhan here is a sample GCCP deployment manifest, we can make this PR as the starting point but we need to discuss more details

File: releases/v1.2/vsphere-paravirtual/cloud-controller-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+          name: ccm-config
+          readOnly: true
+        - mountPath: /etc/kubernetes/guestclusters/tanzukubernetescluster
+          name: ccm-owner-reference
+          readOnly: true
+        - mountPath: /etc/cloud/ccm-provider
+          name: ccm-provider
+          readOnly: true
+      dnsPolicy: ClusterFirst
+      nodeSelector:
```
Comment (lubronzhan) [inline releases/v1.2/vsphere-paravirtual/cloud-controller-manager.yaml:174]: Please remove this

**Comment by lubronzhan** [comment]
Hi @andrewsykim Should the folder of release `1.2` renamed to `1.20`?

File: releases/v1.2/vsphere-paravirtual/cloud-controller-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        - --controllers=cloud-node
+        - --cloud-config=/config/cloud-config
+        - --cluster-name={{ .TanzuKubernetesClusterName }}
+        - --cloud-provider=vsphere-paravirtual
+        env:
+        - name: SUPERVISOR_APISERVER_ENDPOINT_IP
+          value: '{{ .SupervisorMasterEndpointIP }}'
+        - name: SUPERVISOR_APISERVER_PORT
+          value: '{{ .SupervisorMasterPort }}'
+        image: gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.2.1
```
[the remark this answers is not in the document]
Comment (nicolehanjing) [reply releases/v1.2/vsphere-paravirtual/cloud-controller-manager.yaml:148]: I'm not sure actually but maybe @andrewsykim knows?

**Review by lubronzhan — APPROVED** [review]

**Comment by lubronzhan** [comment]
> Hi @andrewsykim Should the folder of release `1.2` renamed to `1.20`?

hm,. I checked release note. Is this release version isolated from K8s version? https://github.com/kubernetes/cloud-provider-vsphere/releases/tag/v1.2.0
But I also see 1.18.1. https://github.com/kubernetes/cloud-provider-vsphere/releases/tag/v1.18.1
Do we want to match k8s or just stick with CP own version?

**Comment by andrewsykim** [comment]
v1.2 was the last version before we started to tag our releases in lock-step with Kubernetes release

**Pull request closed without merging**